src/UhdmAst.cpp

```diff
@@ -1695,6 +1695,51 @@ AstNode* process_typedef(vpiHandle obj_h, UhdmShared& shared) {
     return typedefp;
 }
 
+AstNode* process_typedefs_and_params(vpiHandle obj_h, UhdmShared& shared) {
```

Comment:
This function handles two separate things.
A function to handle parameters that's aware of the `vpiParameter`/`vpiParamAssign` split would be nice to have and reuse. It does not need to touch the typedefs (I'm not sure they warrant a separate function, but feel free to add one if you think it would be clearer).
The name is confusing as well - so far, `process_*` functions work on a VPI node of given type and do not visit multiple types.

Reply:
Done